### PR TITLE
[Messenger] Rename Adapters to Transports

### PR DESCRIPTION
**Pull request by sroze**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ø
| License       | MIT
| Doc PR        | ø

Last of our tasks on the "plan to beta", renaming "adapters" to "transports". This is a term that makes more sense and is commonly used within the "queue community".

### CODE REVIEW

**Review by ogizanagi — APPROVED** [review]
There are still some occurrences of `adapter` in the component. Does it make sense to keep those? It doesn't disturb me at those places, but asking anyway :)

screenshot 2018-05-02 a 22 12 56

**Comment by Tobion** [comment]
What I also found is that namespaces of interfaces and implementations are not consistent.
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Messenger/MiddlewareInterface.php is in the messener root. But the implementations are in https://github.com/symfony/symfony/tree/master/src/Symfony/Component/Messenger/Middleware
But the implemtations of https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Messenger/HandlerLocatorInterface.php are also in the root.
https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Messenger/Handler/MessageHandlerInterface.php is not in the root and the implementation neither.

**Comment by Tobion** [comment]
So I'd propose
1. move the bus stuff to Symfony\Component\Messenger\Bus\
2. move handler locators to Symfony\Component\Messenger\Handler\Locator
3. move Symfony\Component\Messenger\MiddlewareInterface to Symfony\Component\Messenger\Middleware\MiddlewareInterface

**Comment by ogizanagi** [comment]
I agree with @Tobion . But I'd keep `MessageBus` and `MessageBusInterface` root as the main component entrypoint. It seems consistent with what we did with other components.

**Comment by sroze** [comment]
Good catch @Tobion, it was the right time to move them. I've moved the handler and middleware ones, kept the Bus one at the root like we do on other bundles like the Serializer.

**Comment by sroze** [comment]
Note that the failures on the `FrameworkBundle` (for deps=low & deps=high) makes sense, as the class is different in `master` :)

**Review by chalasr — APPROVED** [review]

**Review by weaverryan — APPROVED** [review]
👍 adapters -> transports & the other renamings.

**Review by Tobion — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @sroze.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml [hunk 1]
```diff
@@ -60,16 +60,16 @@
         </service>
 
         <!-- Adapters -->
```
Comment (Koc) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml]: Does this line also should be renamed?

**Pull request merged**